osu.Server.Spectator.Tests/Multiplayer/UserStateManagementTests.cs

```diff
@@ -317,5 +317,37 @@ public async Task CanNotAbortGameplayInNonGameplayStates()
             await Hub.ChangeState(MultiplayerUserState.Ready);
             await Assert.ThrowsAsync<InvalidStateException>(() => Hub.AbortGameplay());
         }
+
+        [Fact]
+        public async Task KickedUserReceivesCallback()
+        {
+            await Hub.JoinRoom(ROOM_ID);
+
+            // Join another user to be kicked.
+            SetUserContext(ContextUser2);
+            await Hub.JoinRoom(ROOM_ID);
+
+            SetUserContext(ContextUser);
+            await Hub.KickUser(USER_ID_2);
+
+            User2Receiver.Verify(r => r.UserLeft(It.IsAny<MultiplayerRoomUser>()), Times.Never);
+            User2Receiver.Verify(r => r.UserKicked(It.IsAny<MultiplayerRoomUser>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task LeftUserReceivesCallback()
+        {
+            await Hub.JoinRoom(ROOM_ID);
+
+            // Join another user to make sure the room doesn't close when the user leaves.
+            SetUserContext(ContextUser2);
+            await Hub.JoinRoom(ROOM_ID);
+
+            SetUserContext(ContextUser);
+            await Hub.LeaveRoom();
+
+            UserReceiver.Verify(r => r.UserLeft(It.IsAny<MultiplayerRoomUser>()), Times.Once);
+            User2Receiver.Verify(r => r.UserKicked(It.IsAny<MultiplayerRoomUser>()), Times.Never);
```

Comment:
Not sure if this is a case of a copy-paste gone wrong - judging from the other test I'd expect to see

```suggestion
            UserReceiver.Verify(r => r.UserKicked(It.IsAny<MultiplayerRoomUser>()), Times.Never);
```

Reply:
Indeed, I copy-pasted this from the line above :sweat_smile: